Source/Core/Common/NandPaths.cpp

```diff
@@ -151,4 +156,11 @@ std::string UnescapeFileName(const std::string& filename)
 
   return result;
 }
+
+bool IsFileNameSafe(const std::string& filename)
+{
+  return !std::all_of(filename.begin(), filename.end(), [](char c) { return c == '.'; }) &&
+         std::none_of(filename.begin(), filename.end(),
+                      [](char c) { return IsIllegalCharacter(c); });
```

Comment:
`IsIllegalCharacter` can be passed directly into the none_of call without the lambda around it, since it fits the criteria of a predicate function as is.

Reply:
Oh yeah, I always forget lambdas work like that